src/components/flat-table/flat-table-header/flat-table-header.component.tsx

```diff
@@ -58,28 +36,27 @@ export const FlatTableHeader = ({
   width,
   py,
   px,
-  reportCellWidth,
-  cellIndex,
-  leftPosition,
-  rightPosition,
+  id,
   ...rest
 }: FlatTableHeaderProps) => {
   const ref = useRef<HTMLTableCellElement>(null);
+  const internalId = useRef(id || guid());
   const { colorTheme } = useContext(FlatTableThemeContext);
 
-  useLayoutEffect(() => {
-    if (ref.current && reportCellWidth) {
-      reportCellWidth(ref.current.offsetWidth, cellIndex);
-    }
-  }, [reportCellWidth, cellIndex]);
+  const { leftPositions, rightPositions } = useContext(FlatTableRowContext);
```

Comment:
**suggestion** (non-blocking): this code block seems to be being used in a few places (reading context values and setting `leftPosition`, `rightPosition` and `makeCellSticky` from them). It might be DRYer as a custom hook?

Reply:
Good idea I'll see what I can do 👍